src/bin/muxer/mod.rs

```diff
@@ -32,18 +32,21 @@ pub trait Muxer: Send {
   fn flush(&mut self) -> io::Result<()>;
 }
 
-pub fn create_muxer(
-  path: &str, overwrite: bool,
+pub fn create_muxer<P: AsRef<Path>>(
+  path: P, overwrite: bool,
 ) -> Result<Box<dyn Muxer + Send>, CliError> {
   if !overwrite {
-    IvfMuxer::check_file(path)?;
+    IvfMuxer::check_file(path.as_ref())?;
   }
 
-  if path == "-" {
-    return IvfMuxer::open(path);
+  if let Some(path) = path.as_ref().to_str() {
+    if path == "-" {
+      return IvfMuxer::open(path);
+    }
   }
 
-  let ext = Path::new(path)
+  let ext = path
+    .as_ref()
     .extension()
     .and_then(OsStr::to_str)
```

Comment:
You can probably drop this conversion to OsStr now.

Reply:
Do you mean the conversion to &str? We use it lower in match. Matching on OsStr is still in rfc.

Reply:
Ah I see.